iModelCore/ECDb/ECDb/SchemaWriter.cpp

```diff
@@ -5236,7 +5247,7 @@ BentleyStatus SchemaWriter::UpdateSchema(Context& ctx, SchemaChange& schemaChang
             return ERROR;
             }
 
-        if (!ctx.AreMajorSchemaVersionChangesAllowed() && !(newSchema.IsDynamicSchema() && ctx.IsMajorSchemaVersionChangeAllowedForDynamicSchemas()))
+        if (!ctx.IsAlwaysAllowDeletionsEnabled() && !ctx.AreMajorSchemaVersionChangesAllowed() && !(newSchema.IsDynamicSchema() && ctx.IsMajorSchemaVersionChangeAllowedForDynamicSchemas()))

```

Comment:
This one here seems not logical to me. The flag does not state that is allows incrementing the read version. There is a separate flag for that already (AreMajowSchemaVersionChangesAllowed) so that side effect seems unnecessary.